examples/scale/scenarios/teamshooter/teamshooter.go

```diff
@@ -0,0 +1,317 @@
+// arguments used:
+// mode: The game mode the players wants to play in. mode is a hard partition.
+// regions: Players may have good latency to one or more regions. A player will
+//   search for matches in all eligible regions.
+// skill: Players have a random skill based on a normal distribution. Players
+//   will only be matched with other players who have a close skill value. The
+//   match functions have overlapping partitions of the skill brackets.
+package teamshooter
+
+import (
+	"fmt"
+	"io"
+	"math"
+	"math/rand"
+	"sort"
+	"time"
+
+	"github.com/golang/protobuf/ptypes"
+	"github.com/golang/protobuf/ptypes/any"
+	"github.com/golang/protobuf/ptypes/wrappers"
+	"open-match.dev/open-match/pkg/pb"
+)
+
+const (
+	poolName = "all"
+	skillArg = "skill"
+	modeArg  = "mode"
+)
+
+// Scenario creates a new TeamShooterScenario.
+func Scenario() *TeamShooterScenario {
+
+	modes, randomMode := weightedChoice(map[string]int{
+		"pl": 100,
+		"cp": 25,
+	})
+
+	regions := []string{}
+	for i := 0; i < 2; i++ {
+		regions = append(regions, fmt.Sprintf("region_%d", i))
+	}
+
+	return &TeamShooterScenario{
+		regions:            regions,
+		maxRegions:         1,
+		playersPerGame:     12,
+		skillBoundaries:    []float64{math.Inf(-1), 0, math.Inf(1)},
+		maxSkillDifference: 0.01,
+		modes:              modes,
+		randomMode:         randomMode,
+	}
+}
+
+// TeamShooterScenario provides the required methods for running a scenario.
+type TeamShooterScenario struct {
+	regions            []string
+	maxRegions         int
+	playersPerGame     int
+	skillBoundaries    []float64
+	maxSkillDifference float64
+	modes              []string
+	randomMode         func() string
+}
+
+// Profiles shards the player base on mode, region, and skill.
+func (t *TeamShooterScenario) Profiles() []*pb.MatchProfile {
+	p := []*pb.MatchProfile{}
+
+	for _, region := range t.regions {
+		for _, mode := range t.modes {
+			for i := 0; i+1 < len(t.skillBoundaries); i++ {
+				skillMin := t.skillBoundaries[i] - t.maxSkillDifference/2
+				skillMax := t.skillBoundaries[i+1] + t.maxSkillDifference/2
+				p = append(p, &pb.MatchProfile{
+					Name: fmt.Sprintf("%s_%s_%v-%v", region, mode, skillMin, skillMax),
+					Pools: []*pb.Pool{
+						{
+							Name: poolName,
+							DoubleRangeFilters: []*pb.DoubleRangeFilter{
+								{
+									DoubleArg: skillArg,
+									Min:       skillMin,
+									Max:       skillMax,
+								},
+							},
+							TagPresentFilters: []*pb.TagPresentFilter{
+								{
+									Tag: region,
+								},
+							},
+							StringEqualsFilters: []*pb.StringEqualsFilter{
+								{
+									StringArg: modeArg,
+									Value:     mode,
+								},
+							},
+						},
+					},
+				})
+			}
+		}
+	}
+
+	return p
+}
+
+// Ticket creates a randomized player.
+func (t *TeamShooterScenario) Ticket() *pb.Ticket {
+	region := rand.Intn(len(t.regions))
+	numRegions := rand.Intn(t.maxRegions) + 1
+
+	tags := []string{}
+	for i := 0; i < numRegions; i++ {
```

Comment:
In the real world, valid regions is going to have a lot of overlap.  This is because players who are close to each other will be close to the same regions.